src/Components/Sidebar/CategoryFilter/CategoryFilter.jsx

```diff
@@ -1,29 +1,50 @@
 import axios from "axios";
 import { useState, useEffect } from "react";
+import { useFilterReducerContext } from "../../../context/FilterReducer.context";
+import { Loader } from "../../Loader/Loader";
 
 export const CartegoryFilter = () => {
+  const { loading, categories, dispatch } = useFilterReducerContext();
+
   const [categoryData, setCategoryData] = useState([]);
 
+  // initializing categories from data
   useEffect(() => {
     (async function () {
       const { data } = await axios.get("/api/categories");
       setCategoryData(data.categories);
     })();
   }, []);
-  
+
   return (
     <>
       <li className="filter-section-title">Categories</li>
-      {categoryData.map((category) => {
-        return (
-          <li key={category.id}>
-            <label className="form-label">
-              <input type="checkbox" />
-              {category.categoryName}
-            </label>
-          </li>
-        );
-      })}
+      {loading ? (
+        <Loader />
+      ) : (
+        <>
+          {categoryData.map((category) => {
```

Comment:
Here, we can remove the return statement. As there is only one element that has to be returned so you can directly write after `=>`

Like,
```javascript
.map((category) => (<li></li>))
```

Reply:
thanks for pointing it out i will make the changes